tests/test_io_utils.py

```diff
@@ -14,35 +15,61 @@
     "filename, mean, shape",
     [
         ("img.tif", 13577.494856, (2048, 2048, 3)),
-        ("img.jpg", 52.412425, (2048, 2048, 3)),
         ("img.png", 52.33736, (2048, 2048, 3)),
+        ("img.jpg", 52.412425, (2048, 2048, 3)),
         ("img-8bit.jp2", 47.09060, (343, 343, 3)),
         ("img-15bit.jp2", 0.3041897, (1830, 1830)),
         ("img-16bit.jp2", 0.3041897, (1830, 1830)),
     ],
 )
-def test_img_read(input_folder: str, output_folder: str, filename: str, mean: float, shape: Tuple[int, ...]) -> None:
-    is_jpeg = filename.endswith("jpg")
+def test_img_read(input_folder: str, filename: str, mean: float, shape: Tuple[int, ...]) -> None:
     img = read_data(os.path.join(input_folder, filename))
 
     assert img.shape == shape
+    assert np.mean(img) == approx(mean, abs=1e-4)
+    assert img.flags["WRITEABLE"], "Obtained numpy array is not writeable"
 
-    if is_jpeg or python_implementation() != "PyPy":
-        assert np.mean(img) == approx(mean, abs=1e-4)
 
-    assert img.flags["WRITEABLE"], "Obtained numpy array is not writeable"
+xml_root = ET.Element("EOPatch")
+xml_data = ET.SubElement(xml_root, "data")
+ET.SubElement(xml_data, "field1", name="BANDS-S2-L1C").text = "some value1"
+ET.SubElement(xml_data, "field2", name="CLP").text = "some value2"
+TEST_XML = ET.ElementTree(xml_root)
 
-    file_path = os.path.join(output_folder, filename)
 
-    if is_jpeg:
+@pytest.mark.parametrize(
+    "filename, data",
+    [
+        ("img.tif", np.arange(5 * 5 * 3).reshape((5, 5, 3))),
+        ("img.png", np.arange((5 * 5 * 3), dtype=np.uint8).reshape((5, 5, 3))),
+        ("img-8bit.jp2", np.arange((5 * 5 * 3), dtype=np.uint8).reshape((5, 5, 3))),
+        ("img-15bit.jp2", np.arange((5 * 5 * 3), dtype=np.uint8).reshape((5, 5, 3))),
+        ("img-16bit.jp2", np.arange((5 * 5 * 3), dtype=np.uint8).reshape((5, 5, 3))),
+        ("test-string.txt", "sentinelhub-py is often shortened to sh-py"),
+        ("test-xml.xml", TEST_XML),
+    ],
+)
+def test_write(filename: str, data: Union[str, np.ndarray, ET.ElementTree]) -> None:
```

Comment:
i'd perhaps rename it to `test_write_read` :thinking: 
Also do you think it could be worthwhile to check `filesystem.exists(filename)`?

Reply:
the thing is that this check checks working of fs.tempfs 
and is kinda move in other direction from micro tests
but i can add it it one line of code that's relay fast

Reply:
done